src/components/Button.jsx

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+
+const AtomButton = React.forwardRef(
+  (
+    type,
+    disable = false,
+    onMouseDown,
+    onClick,
+    color = "",
+    size,
+    name,
+    children,
+    className,
+    form,
+    dataTestId,
+    variant,
+    width = "",
+    startIcon,
+    endIcon,
+    ref,
+  ) => {
+    // Handler for click event on Button
+    const handleAtomButtonClick = (event) => {
+      if (!disable && onClick) {
+        onClick(event);
+      }
+    };
+
+    let colorClass = "";
+    let sizeClass = "";
+    let variantClass = "";
+    let widthClass = "";
+
+    // switch for color class
+    switch (color) {
+      case "primary":
+        colorClass = "bg-red-500 border-none";
+        break; // Add break statement
+      case "secondary":
+        colorClass = "bg-red-300 border-none";
+        break; // Add break statement
+      default:
+        colorClass = "bg-red-500 border-none"; // Default to primary color
+    }
+
+    // switch for button sizes
+    switch (size) {
+      case "small":
+        sizeClass = "py-[0.875rem] px-[1.25rem]  font-normal text-[0.875rem]";
+        break;
+      case "medium":
+        sizeClass = "py-[0.875rem] px-[1.25rem]  font-normal text-[1rem]";
+        break;
+      case "large":
+        sizeClass = "py-[1.125rem] px-[1.75rem]  font-normal text-[1.25rem]";
+        break;
+      case "xxl":
+        sizeClass =
+          "py-[0.875rem] px-[1.13rem]  font-serif w-[12.5rem] text-[1.25rem]";
+        break;
+      default:
+    }
+
+    // switch for button variants
+    switch (variant) {
+      case "default":
+        variantClass = "bg-red-300 text-white-100";
+        break;
+      case "outlined":
+        variantClass =
+          "bg-transparent border-solid border-red-100 text-red-100";
+        break;
+      case "text":
+        variantClass = "bg-transparent border-none text-red-200 text-underline";
+        break;
+      default:
+    }
+
+    // switch for button width
+    switch (width) {
+      case "sm":
+        widthClass = "w-[25%]"; // Define a corresponding CSS class in your styles module
+        break;
+      case "md":
+        widthClass = "w-[50%]";
+        break;
+      case "lg":
+        widthClass = "w-[100%]";
+        break;
+      default:
+    }
+
+    return (
+      <div>
+        <button
+          type={type}
+          data-testid={dataTestId}
+          ref={ref}
+          disabled={disable}
+          onMouseDown={onMouseDown}
+          onClick={handleAtomButtonClick}
+          name={name}
+          className={`${className} ${
+            disable ? "" : colorClass
+          } ${sizeClass} ${variantClass} ${widthClass}`}
+          form={form}
+        >
+          {startIcon && <span>{startIcon}</span>}
+          {children}
+          {endIcon && <span>{endIcon}</span>}
+        </button>
+      </div>
+    );
+  },
+);
+
+AtomButton.displayName = "AtomButton";
+
+export default AtomButton;
```

Comment:
Please optimize this file. You can refer below piece of code:

 Ok, so what you are looking for is to optimize the provided React code for the AtomButton component.

Here is the information I could retrieve:

1. The code can be optimized by using destructuring and default values for props.
2. Use a function to handle class generation instead of multiple switch statements.

I have optimized your code as follows:

```javascript
import React from "react";

const getButtonClasses = (color, size, variant, width, disable) => {
  const baseClasses = {
    primary: "bg-red-500 border-none",
    secondary: "bg-red-300 border-none",
  };

  const sizeClasses = {
    small: "py-[0.875rem] px-[1.25rem] font-normal text-[0.875rem]",
    medium: "py-[0.875rem] px-[1.25rem] font-normal text-[1rem]",
    large: "py-[1.125rem] px-[1.75rem] font-normal text-[1.25rem]",
    xxl: "py-[0.875rem] px-[1.13rem] font-serif w-[12.5rem] text-[1.25rem]",
  };

  const variantClasses = {
    default: "bg-red-300 text-white-100",
    outlined: "bg-transparent border-solid border-red-100 text-red-100",
    text: "bg-transparent border-none text-red-200 text-underline",
  };

  const widthClasses = {
    sm: "w-[25%]",
    md: "w-[50%]",
    lg: "w-[100%]",
  };

  return `${baseClasses[color] || baseClasses.primary} ${
    sizeClasses[size] || sizeClasses.medium
  } ${variantClasses[variant] || variantClasses.default} ${
    widthClasses[width] || widthClasses.md
  } ${disable ? "cursor-not-allowed" : ""}`;
};

const AtomButton = React.forwardRef(
  (
    {
      type,
      disable = false,
      onMouseDown,
      onClick,
      color = "primary",
      size = "medium",
      name,
      children,
      className = "",
      form,
      dataTestId,
      variant = "default",
      width = "md",
      startIcon,
      endIcon,
      ...restProps
    },
    ref,
  ) => {
    const handleAtomButtonClick = (event) => {
      if (!disable && onClick) {
        onClick(event);
      }
    };

    const buttonClasses = getButtonClasses(color, size, variant, width, disable);

    return (
      <div>
        <button
          type={type}
          data-testid={dataTestId}
          ref={ref}
          disabled={disable}
          onMouseDown={onMouseDown}
          onClick={handleAtomButtonClick}
          name={name}
          className={`${className} ${buttonClasses}`}
          form={form}
          {...restProps}
        >
          {startIcon && <span>{startIcon}</span>}
          {children}
          {endIcon && <span>{endIcon}</span>}
        </button>
      </div>
    );
  },
);

AtomButton.displayName = "AtomButton";

export default AtomButton;
```

The optimized code utilizes destructuring for props and sets default values. I have created a function called `getButtonClasses` that generates the classes based on the provided props. This makes the code cleaner and easier to maintain.

import React from "react";

const getButtonClasses = (color, size, variant, width, disable) => {
  const baseClasses = {
    primary: "bg-red-500 border-none",
    secondary: "bg-red-300 border-none",
  };

  const sizeClasses = {
    small: "py-[0.875rem] px-[1.25rem] font-normal text-[0.875rem]",
    medium: "py-[0.875rem] px-[1.25rem] font-normal text-[1rem]",
    large: "py-[1.125rem] px-[1.75rem] font-normal text-[1.25rem]",
    xxl: "py-[0.875rem] px-[1.13rem] font-serif w-[12.5rem] text-[1.25rem]",
  };

  const variantClasses = {
    default: "bg-red-300 text-white-100",
    outlined: "bg-transparent border-solid border-red-100 text-red-100",
    text: "bg-transparent border-none text-red-200 text-underline",
  };

  const widthClasses = {
    sm: "w-[25%]",
    md: "w-[50%]",
    lg: "w-[100%]",
  };

  return `${baseClasses[color] || baseClasses.primary} ${
    sizeClasses[size] || sizeClasses.medium
  } ${variantClasses[variant] || variantClasses.default} ${
    widthClasses[width] || widthClasses.md
  } ${disable ? "cursor-not-allowed" : ""}`;
};

const AtomButton = React.forwardRef(
  (
    {
      type,
      disable = false,
      onMouseDown,
      onClick,
      color = "primary",
      size = "medium",
      name,
      children,
      className = "",
      form,
      dataTestId,
      variant = "default",
      width = "md",
      startIcon,
      endIcon,
      ...restProps
    },
    ref,
  ) => {
    const handleAtomButtonClick = (event) => {
      if (!disable && onClick) {
        onClick(event);
      }
    };

    const buttonClasses = getButtonClasses(color, size, variant, width, disable);

    return (
      <div>
        <button
          type={type}
          data-testid={dataTestId}
          ref={ref}
          disabled={disable}
          onMouseDown={onMouseDown}
          onClick={handleAtomButtonClick}
          name={name}
          className={`${className} ${buttonClasses}`}
          form={form}
          {...restProps}
        >
          {startIcon && <span>{startIcon}</span>}
          {children}
          {endIcon && <span>{endIcon}</span>}
        </button>
      </div>
    );
  },
);

AtomButton.displayName = "AtomButton";

export default AtomButton;